### ISSUE TITLE
Compilation errors when using test package in istio 1.1.x

### ISSUE THREAD

**Issue by PhilipGough** [istio/istio#11226]
**Describe the bug**
We have an existing out-of-process adapter which currently, has a dependency on istio `1.0.5` in `Gopkg.toml`. The code uses istio's built in framework for integration testing.

Attempting to upgrade the dependency to any of the 1.1.x releases or branch and running the same integration test results in compilation errors:

``` bash
vendor/k8s.io/client-go/discovery/restmapper.go:42:75: undefined: meta.VersionInterfacesFunc
vendor/k8s.io/client-go/discovery/restmapper.go:176:19: undefined: meta.VersionInterfacesFunc
vendor/k8s.io/client-go/dynamic/dynamic_util.go:31:47: undefined: meta.VersionInterfaces
vendor/k8s.io/client-go/dynamic/dynamic_util.go:32:10: undefined: meta.VersionInterfaces
vendor/k8s.io/client-go/dynamic/dynamic_util.go:39:78: undefined: meta.VersionInterfacesFunc
vendor/k8s.io/client-go/dynamic/dynamic_util.go:40:33: too many arguments in call to meta.NewDefaultRESTMapper
vendor/istio.io/istio/pkg/mcp/client/client.go:126:14: undefined: v1alpha1.Client
vendor/istio.io/istio/pkg/mcp/server/server.go:95:15: undefined: v1alpha1.Envelope
```

No other direct dependencies (or sub-deps from what I can see), depend on k8's client-go

**Expected behavior**
Expect to be able to add istio v1.1.x as a dependency, run `dep ensure` and be able to use `"istio.io/istio/mixer/pkg/adapter/test"` package without compile time errors.

**Version**
Istio version `1.1.x`
OpenShift 3.11

**Comment by PhilipGough** [comment]
Looks like a breaking change here https://github.com/istio/istio/pull/10628

**Comment by hzxuzhonghu** [comment]
It is likely caused by k8s.io/client-go upgrade

**Comment by PhilipGough** [comment]
Well, the Gopkg.toml references branch names of its own dependencies. [client-go](https://github.com/istio/istio/blob/release-1.1/Gopkg.toml#L52) for example. So it's not really surprising this is broken.

That method works fine for istio itself since it uses a lock file and vendors but for external code that depends on istio, this is a major problem.

**Comment by PhilipGough** [comment]
@kyessenov pinging you just fyi. Anyone who attempts to start an OOP adapter currently based off 1.1 will hit this. Is it intended that this path is supported? If so, maybe worth linking to this workaround.

**Comment by kyessenov** [comment]
Thanks for letting us know. We did not have strong guarantees on updates of mixer's test framework. However, it is certainly true that it's being used for out-of-process adapter development. I think the step forward for us is to isolate the test framework into something less dependent on the rest of istio. That way accidental dependency updates in istio are not going to break adapter developers.

/cc @geeknoid for visibility.

**Comment by PhilipGough** [comment]
Just for an update this is essentially fixed now with the 1.1.0 release. I did need to add an override for api like so, since it points at the branch as opposed to the tag:

```[[override]]
  name = "istio.io/api"
  version = "1.1.0"```

**Comment by howardjohn** [comment]
@PhilipGough mentioned this is fixed, so I think this is safe to close. If similar issues exist the cause is likely unrelated as we have moved to go modules rather than dep. Feel free to open new issues or reopen if problems still exist
